app/models/user.rb

```diff
@@ -10,8 +10,13 @@ class User < ApplicationRecord
   before_validation :get_coords
 
   def self.search_for_skills(query)
-    skills = query.split(',').map { |skill| skill.strip.downcase}
-    User.joins(:skills).where("LOWER(skills.name) IN (?)", skills).distinct
+    skills = query.split(',').map { |skill| skill.strip.downcase }
+    
+    conditions = skills.map { |skill|
+      "LOWER(skills.name) ILIKE '%#{sanitize_sql_like(skill)}%'"
+    }.join(' OR ')
```

Comment:
Does sanitize_sql_like have any kind of protection against injection? I ask just because I know AR usually warns against string interpolating anything users can pass into a query.

Reply:
https://api.rubyonrails.org/v7.1.1/classes/ActiveRecord/Sanitization/ClassMethods.html

that's my understanding of it - the docs say "safe" which I assumed meant that. But I asked the AI overlords as well and it stated being used explicitly to avoid injections threats. So the combo of the docs, and AI supporting it tells me that it does exactly what your expressed in your concerns.